src/other/index.jl

```diff
@@ -226,6 +226,10 @@ end
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
 @inline Base.getindex(x::AbstractIndex, notidx::Not) =
     setdiff(1:length(x), getindex(x, notidx.skip))
+@inline Base.getindex(x::AbstractIndex, notidx::Not{<:AbstractVector}) =
```

Comment:
@nalimilan - change added. Now passing duplicates like `df[:, Not([:a, :a, :b]))` will not throw an error.